keps/sig-apps/4368-support-managed-by-label-for-batch-jobs/README.md

```diff
@@ -0,0 +1,1098 @@
+upstream.
+
+First, this risk, to some extent, exists even today as admins with access to the
+control plane can disable job controller by passing `--controllers=-job,*` in the manifest for
+`kube-controller-manager` (see more info on the `--controllers` flag
+[here](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/)).
+
+Second, we believe that users who had the need to fork the Job controller
+already introduced dedicated Job CRDs for their needs.
+
+#### Two controllers running at the same time on old version
+
+It is possible that one configures MultiKueue (or another project) with an
+older version of k8s which does not support the label yet. In that case
+two controllers might start running and compete with Job status updates at the
+same time.
+
+Note that an analogous situation may happen when the version of Kubernetes
+already supports the label, but the feature gate is disabled in `kube-controller-manager`.
+
+To mitigate this risk we warn about it in Kueue documentation, to use this label
+only against newer versions of Kubernetes.
+
+Finally, this risk will fade away with time as the new versions of
+Kubernetes support it.
+
+#### Debuggability
+
+With this mechanism new failure modes can occur. For example, a user may make
+a typo in the label value, or the cluster administrator may not install the
+custom controller (like MultiKueue) on that cluster.
+
+In such cases the user may not observe any progress by the job for a long time
+and my need to debug the Job.
+
+In order to allow for debugging of situations like this the Job controller will
+put a log line indicating the synchronization is delegated to another controller
+(see [implementation overview](#implementation-overview)).
+
+Additionally, re-evaluate extending the `kubectl` command-line tool
+before [GA](#ga). We could extend the command to provide useful debugging
+information with the following:
+- new `MANAGED_BY` column for `kubectl get job -owide` (possibly also without `-owide`)
+- a line in the `kubectl describe job` output, just before the list of events,
+providing a user readable information if the Job is synchronized by a custom
+controller.
+
+Alternative ideas considered were
+[a dedicated condition](#condition-to-indicated-job-is-skipped)
+and [events](#event-indicating-the-job-is-skipped).
+
+#### Custom controllers not compatible with API assumptions by CronJob
+
+Currently, the validation of the Job status API is rather relaxed, allowing for
+potential situations, when modified by custom controllers, which might be
+problematic for the API consumers, like the built-in CronJob controller.
+
+In particular, the CronJob relies on the assumption that the `CompletionTime`
+is only set to successful jobs. Also, that a job does not flip its state from
+Complete to Failed (or the other way round). Also, a finished job should not
+flip back to non-finished.
```

Comment:
agree on strengthening the validation to prevent invalid transitions like this

Reply:
Strong :+1:, if we're planning to allow external controllers update the status, we need to ensure that job won't start flapping between states.